packages/browser-sdk/src/auto-confirm/autoConfirmDisable.test.ts

```diff
@@ -0,0 +1,50 @@
+import { autoConfirmDisable } from "./autoConfirmDisable";
+import { getProvider } from "./getProvider";
+import type { PhantomProvider } from "./types";
+
+jest.mock("./getProvider", () => ({
+  getProvider: jest.fn(),
+}));
+
+const mockGetProvider = getProvider as jest.MockedFunction<() => PhantomProvider | null>;
+
+describe("autoConfirmDisable", () => {
+  let mockProvider: Partial<PhantomProvider>;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockProvider = {
+      request: jest.fn(),
+    };
+    mockGetProvider.mockReturnValue(mockProvider as PhantomProvider);
+  });
+
+  it("should disable auto-confirm", async () => {
+    const mockResult = { enabled: false, chains: [] };  
+    (mockProvider.request as jest.Mock).mockResolvedValue(mockResult);
```

Comment:
Nit: if we alter the method, the type/response will always be correct. What do you think about explicitly typing a more 'true' implementation? e.g.

```suggestion
    const mockResult: AutoConfirmResult = { enabled: false, chains: [] };  
    (mockProvider.request as jest.Mock).mockImplementation((method) => {
      if (method === 'phantom_auto_confirm_disable') {
        return Promise.resolve(mockResult)
      }
      throw new Error(`Unknown ${method}`)
    );
```

We could also do this for every test/suite but I won't mark the others.